src/datachain/lib/video.py

```diff
@@ -0,0 +1,185 @@
+import posixpath
+import shutil
+import tempfile
+from typing import Optional
+
+from numpy import ndarray
+
+from datachain.lib.file import (
+    FileError,
+    ImageFile,
+    Video,
+    VideoFile,
+    VideoFragment,
+    VideoFrame,
+)
+
+try:
+    import ffmpeg
+    import imageio.v3 as iio
+except ImportError as exc:
+    raise ImportError(
+        "Missing dependencies for processing video.\n"
+        "To install run:\n\n"
+        "  pip install 'datachain[video]'\n"
+    ) from exc
+
+
+def video_info(file: VideoFile) -> Video:
+    """
+    Returns video file information.
+
+    Args:
+        file (VideoFile): Video file object.
+
+    Returns:
+        Video: Video file information.
+    """
+    try:
+        probe = ffmpeg.probe(file.get_local_path())
+    except ffmpeg.Error as exc:
+        raise FileError(file, f"unable to probe video file: {exc.stderr}") from exc
+    except Exception as exc:
+        raise FileError(file, f"unable to probe video file: {exc}") from exc
+
+    if not probe:
+        raise FileError(file, "unable to probe video file")
+
+    all_streams = probe.get("streams")
+    video_format = probe.get("format")
+    if not all_streams or not video_format:
+        raise FileError(file, "unable to probe video file")
+
+    video_streams = [s for s in all_streams if s["codec_type"] == "video"]
+    if len(video_streams) == 0:
+        raise FileError(file, "no video streams found in video file")
+
+    video_stream = video_streams[0]
```

Comment:
[Q] Why take the first one? Is there only ever one?

Reply:
Some video container formats (such as MPEG-4) support multiple video, audio, and subtitle streams. However, in practice, video files with multiple video streams are extremely rare. Most applications, including video players and streaming platforms, are not designed to handle multiple video streams because there is little practical use for them.

I have heard of a few specialized use cases, such as movies with different aspect ratios or sports event recordings featuring multiple camera angles. However, these are rare exceptions, and I have never encountered them in real-world scenarios. In most cases, it is much simpler to provide multiple separate video files and allow users to download or stream only the one they need.